### PR TITLE
Add endpoint to list existing agents

### PR DESCRIPTION
**Pull request by jpcaparas**
## Change log
Add endpoint to list existing agents.

## Tests
Run `./vendor/bin/codecept run functional` on the project root to run functional tests.

### CODE REVIEW

File: app/Http/Controllers/Api/Agent/ListController.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+namespace App\Http\Controllers\Api\Agent;
+
+use App\Http\Controllers\Controller;
+use App\Models\Agent;
+
+class ListController extends Controller
+{
+    public function __invoke()
+    {
+        $agents = Agent::paginate();
```
Comment (titopixelfusion) [inline app/Http/Controllers/Api/Agent/ListController.php]: Don't we need to pass parameter for the page and limit? How do we configure how many elements per page are being displayed?

File: tests/functional/AgentCreateCept.php [hunk 2]
```diff
@@ -14,4 +14,11 @@
 $I->seeResponseContainsJson(array('hostname' => 'example.org'));
 $I->seeResponseContainsJson(array('distro' => 'debian'));
 $I->seeResponseContainsJson(array('version' => '2.1'));
-$I->seeInDatabase('agents', array('hostname' => 'example.org', 'distro' => 'debian'));
```
Comment (titopixelfusion) [inline tests/functional/AgentCreateCept.php]: We shouldn't reuse this file for the test, each end point should have it's own file(s). Notice how misleading it is to have a file name AgentCreate and we have tests for listing in it. I'd suggest to have AgentListCept.php

Comment (jpcaparas) [inline tests/functional/AgentCreateCept.php]: Okay, I'll put it on a separate `-Cept.php` file.

File: app/Http/Controllers/Api/Agent/ListController.php (hunk 1 above)
Comment (jpcaparas) [inline app/Http/Controllers/Api/Agent/ListController.php]: Ideally, the `per_page` (_result limit_) attribute is set at the model level. 

We can pass on a query parameter called `page` that sets the page number (conforming to how Laravel's [pagination results](https://laravel.com/docs/5.4/pagination#displaying-pagination-results) works), but as for the _result limit_ attribute, do we also give the freedom to set this as a query string parameter too?

Comment (jpcaparas) [inline app/Http/Controllers/Api/Agent/ListController.php]: Actually, I just checked, and even without parameters, it already infers the page from the query string

screen shot 2017-03-22 at 10 21 32 am

Comment (titopixelfusion) [inline app/Http/Controllers/Api/Agent/ListController.php]: That's great @jpcaparas , let's add that attribute to the model to be explicit. I had no idea it existed. No need to make it dynamic.  We enforce whatever we set in the model. Maybe set it to 10, not sure what's the default.

Comment (jpcaparas) [inline app/Http/Controllers/Api/Agent/ListController.php]: @titopixelfusion, the default is `15`, inherited from `Illuminate/Database/Eloquent/Model` 

screen shot 2017-03-22 at 2 00 50 pm

I'll set it explicitly on the model.

File: tests/functional/AgentCreateCept.php (hunk 2 above)
Comment (jpcaparas) [inline tests/functional/AgentCreateCept.php]: Hi @titopixelfusion, I've now moved this to its own separate test.

**Comment by jpcaparas** [comment]
@ThijsBouwes @shanemoses-pixelfusion looking forward to your approval / comments. Thanks.

**Comment by jpcaparas** [comment]
@titopixelfusion,

I'll also add this to the Swagger docs once we formalise PR #11

**Pull request merged**